addons/repair/models/repair.py

```diff
@@ -236,14 +236,32 @@ def onchange_partner_id(self):
             self.partner_invoice_id = addresses['invoice']
             self.pricelist_id = self.partner_id.property_product_pricelist.id
 
-    @api.depends('company_id')
+    @api.depends('company_id', 'picking_id', 'lot_id')
     def _compute_location_id(self):
         for order in self:
-            if order.company_id:
-                if order.location_id.company_id != order.company_id:
-                    warehouse = self.env['stock.warehouse'].search([('company_id', '=', order.company_id.id)], limit=1)
-                    order.location_id = warehouse.lot_stock_id
-            else:
+            if order.company_id and order.location_id.company_id != order.company_id:
+                warehouse = self.env['stock.warehouse'].search([('company_id', '=', order.company_id.id)], limit=1)
+                order.location_id = warehouse.lot_stock_id
+            if order.picking_id:
+                order.location_id = order.picking_id.location_dest_id
+            if order.lot_id:
+                quant_id = self.env['stock.quant'].search([
+                    ('product_id', '=', order.product_id.id),
+                    ('lot_id', '=', order.lot_id.id),
+                    ('quantity', '>', 0),
+                    ('location_id.usage', 'in', ['internal','transit'])
+                ], limit=1)
+                if quant_id:
+                    order.location_id = quant_id.location_id
+                else:
+                    move_line = self.env['stock.move.line'].search([
+                        ('product_id', '=', order.product_id.id),
+                        ('lot_id', '=', order.lot_id.id),
+                        ('state', '=', 'done')
+                    ], order='id desc', limit=1)
+                    if move_line:
+                        order.location_id = move_line.location_dest_id
+            if not order.location_id:
                 order.location_id = False
```

Comment:
:crying_cat_face: please do some exercises to better understand when and how you should use if/elif statements: https://docs.python.org/3/tutorial/controlflow.html#if-statements

```suggestion
            if order.lot_id and order.lot_id.quant_ids:
                quants = order.lot_id.quant_ids.filtered(lambda q: q.quantity > 0 and location_id.usage in ['internal', 'customer', 'transit'])
                if len(quants.location_id) == 1:
                    order.location_id = order.lot_id.quant_ids..location_id
            elif order.picking_id:
                order.location_id = order.picking_id.location_dest_id
            elif order.company_id:
                warehouse = self.env['stock.warehouse'].search([('company_id', '=', order.company_id.id)], limit=1)
                order.location_id = warehouse.lot_stock_id
            else:
                order.location_id = False
```